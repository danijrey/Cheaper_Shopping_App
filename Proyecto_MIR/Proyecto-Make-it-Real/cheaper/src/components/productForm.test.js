import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import ProductForm from './productForm';


describe ('ProductForm',() =>{
    it ('should render correctly', ()=>{
    const {container} = render (
    <MemoryRouter>
        <ProductForm/>
    </MemoryRouter> 
    
    ); 
    expect (container.firstChild).toMatchSnapshot();
    });
    it('Should render the element with data-testid="productForm"', () => {
        const { getByTestId } = render(
            <MemoryRouter>
                <ProductForm/>
            </MemoryRouter>
        );
        expect(getByTestId('productForm')).toBeInTheDocument();
    });
    it('Should render the element with the following text="Nuevo","Eliminar","Editar"', () => {
        const {getByText } = render(
            <MemoryRouter>
                <ProductForm/>
            </MemoryRouter>
        );
        expect(getByText(/Nuevo/,/Eliminar/,/Editar/)).toBeInTheDocument();
    })
        it('Should render the element with the following placeholder text="Ingresa","Descripcion","$"', () => {
        const {getByPlaceholderText } = render(
            <MemoryRouter>
                <ProductForm/>
            </MemoryRouter>
        );
        expect(getByPlaceholderText(/Ingresa/,/Descripcion/,/$/)).toBeInTheDocument();
    })
    
});