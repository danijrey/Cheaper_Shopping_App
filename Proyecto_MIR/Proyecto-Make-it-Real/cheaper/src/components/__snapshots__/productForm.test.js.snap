// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductForm should render correctly 1`] = `
<div
  class="ProductDescription"
>
  <div
    class="NavBusqueda-NewProduct"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-search fa-w-16 NavBusqueda-Icon"
      data-icon="search"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        fill="currentColor"
      />
    </svg>
    <input
      class="NavBusqueda-Busqueda"
      placeholder=" Realizar una busqueda."
      type="text"
    />
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-camera fa-w-16 NavBusqueda-Icon"
      data-icon="camera"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 144v288c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V144c0-26.5 21.5-48 48-48h88l12.3-32.9c7-18.7 24.9-31.1 44.9-31.1h125.5c20 0 37.9 12.4 44.9 31.1L376 96h88c26.5 0 48 21.5 48 48zM376 288c0-66.2-53.8-120-120-120s-120 53.8-120 120 53.8 120 120 120 120-53.8 120-120zm-32 0c0 48.5-39.5 88-88 88s-88-39.5-88-88 39.5-88 88-88 88 39.5 88 88z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="Ubicacion"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-map-marker-alt fa-w-12 Ubicacion-Icon"
      data-icon="map-marker-alt"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
        fill="currentColor"
      />
    </svg>
    <input
      class="Ubicacion-Busqueda"
      placeholder=" UbicaciÃ³n."
      type="text"
    />
  </div>
  <div
    class="buttons"
  >
    <a
      href="/products/edit/:id"
    >
      <button
        class="edit"
      >
        Editar producto
      </button>
    </a>
    <a
      href="/products/delete/:id"
    >
      <button
        class="delete"
      >
        Eliminar producto
      </button>
    </a>
    <a
      href="/products/create"
    >
      <button
        class="create"
      >
        Nuevo producto
      </button>
    </a>
  </div>
  <form
    class="ProductsForm"
    data-testid="productForm"
  >
    <div
      class="title"
    >
      <label
        for="name"
      >
        Nombre del producto
      </label>
      <input
        id="name"
        name="name"
        placeholder="Ingresa el nombre del producto"
        type="text"
        value=""
      />
    </div>
    <div
      class="container-img"
    >
      <input
        accept="image/*"
        class="file-img"
        id="picture"
        name="picture"
        type="file"
      />
    </div>
    <textarea
      class="description-text"
      cols="35"
      id="description"
      name="description"
      placeholder="Descripcion del producto"
      rows="5"
    />
    <div
      class="description-article  "
    >
      <div
        class="price-gral"
      >
        <input
          class="price"
          id="price"
          name="price"
          placeholder="$"
          type="text"
          value=""
        />
        <h6>
          Precio
        </h6>
      </div>
    </div>
    <button
      class="create-product"
      type="submit"
    >
      Publicar
    </button>
  </form>
</div>
`;
